extpsdk/README.md

```diff
@@ -0,0 +1,260 @@
+variable instead):
+```
+authToken=XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
+sources=MySourceName
+targetServer=https://dev.vantiq.com/
+sendPings=true
+```
+
+### Program Flow
+
+1. Create a `VantiqConnectorSet`. This will read the config file and establish (but not start) a `VantiqSourceConnection` for each source listed in the config file.
+2. Define the handlers to be used.  You can use a single set of handlers for all sources (using `VantiqConnectorSet.configure_handlers_for_all()`, or you can establish handlers for each source (by iterating through the connections provided by `VantiqConnectorSet.get_connections()` and then calling `VantiqSourceConnection.configure_handlers()` for each connection.
+3. Once the environment is set, you can start the connector operations.  This is done by calling `VantiqConnectorSet.run_connectors()`.  Once this call is made (and awaited), the SDK will take over management of the connections.
+4. Each source connection can perform its work by responding to calls to the `publish` and `query` handlers.
+5. If you need to send an event to the server, use `VantiqSourceConnection.send_notification()`.
+
+If the connection closes, the SDK will automatically re-initiate the connection.  This will result in calls the `connect` handlers again, etc.  Such a reconnection can happen due to network or connection issues.  It can also happen if the source configuration on the Vantiq side is changed.  A changed configuration requires a that the connector obtain the new configuration, and that is done by simply making a new connection.
+
+
+	
+## <a name="VantiqSourceConnection" id="sourceConnection"></a>Using ExtensionWebSocketClient
+Every `VantiqSourceConnection` deals with a single source via its own WebSocket connection. 
+
+### Setup
+
+To operate, each `VantiqSourceConnection` must be configured with a set of handlers.  These handlers configured as follows.  (Note that a similar call can be made using the `VantiqConnectorSet` if all connections should use the same handler.
+
+Call `VantiqSourceConnection.configure_handlers(close_handler, connect_handler, publish_handler, query_handler)`.  These handlers are defined as follows.
+
+* handle_close : Callable[[dict], Awaitable[None]]
+* handle_connect : Callable[[dict, dict], Awaitable[None]
+* handle_publish : Callable[[dict, dict], Awaitable[None]
+* handle_query : Callable[[dict, dict], Awaitable[None]
+
+These parameters specify the callbacks to handle close, connect, publish, and query calls, respectively. For each, the first parameter is a dict providing the context for the operation.  The context will contain an entry keyed with 'source_name' whose value is the name of the source for which this is a callback.  For query calls, there will be an additional entry keyed with 'response_address'. The value here is used to route the query response back to the querier in the Vantiq server.
```

Comment:
You should probably use the `VantiqConnector` constants here, instead of their values.
```suggestion
These parameters specify the callbacks to handle close, connect, publish, and query calls, respectively. For each, the first parameter is a dict providing the context for the operation.  The context will contain an entry keyed with `VantiqConnector.SOURCE_NAME` whose value is the name of the source for which this is a callback.  For query calls, there will be an additional entry keyed with `VantiqConnector.RESPONSE_ADDRESS`. The value here is used to route the query response back to the querier in the Vantiq server.
```